lib/roo/excelx/cell/date.rb

```diff
@@ -20,7 +20,7 @@ def initialize(value, formula, excelx_type, style, link, base_date, coordinate)
         def create_datetime(_,_);  end
 
         def create_date(base_date, value)
-          base_date + value.to_i
+          base_date + value.to_f.round(6)
```

Comment:
How about `value.to_f.round(6).to_i` ?

Reply:
Agree, that should be most convenient for users (except changing to DateTIme, but that will be breaking changes)